clients/src/main/java/org/apache/kafka/common/requests/HeartbeatResponse.java

```diff
@@ -57,6 +57,10 @@ public Errors error() {
         return Errors.forCode(data.errorCode());
     }
 
+    public ByteBuffer userDatas() {
```

Comment:
This is the serialized map of MemberId -> UserData . I was reluctant to deserialize it eagerly for performance reasons. Not sure if this is reasonable or not.

I guess even if we want it to be lazy, we should encapsulate the deserialization and just cache the result.